File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Mapper.Entity/DocumentUnits/Implementation/DocumentUnitMapper.cs
using VecompSoftware.DocSuiteWeb.Entity.DocumentUnits;

namespace VecompSoftware.DocSuiteWeb.Mapper.Entity.DocumentUnits
{
    public class DocumentUnitMapper : BaseEntityMapper<DocumentUnit, DocumentUnit>, IDocumentUnitMapper
    {
        public DocumentUnitMapper()
        { }

        public override DocumentUnit Map(DocumentUnit entity, DocumentUnit entityTransformed)
        {
            #region [ Base ]
            entityTransformed.DocumentUnitName = entity.DocumentUnitName;
            entityTransformed.Year = entity.Year;
            entityTransformed.Number = entity.Number;
            entityTransformed.Subject = entity.Subject;
            entityTransformed.Title = entity.Title;
            entityTransformed.Environment = entity.Environment;
            entityTransformed.Status = entity.Status;
            #endregion

            return entityTransformed;
        }

    }
}

File: SOURCE_EXTERNAL/VecompSoftware.DocSuite.ExternalViewer/src/app/mappers/document-units/document-unit.mapper.ts
import { Injectable } from '@angular/core'; 

import { BaseMapper } from '../base.mapper'; 
import { DocumentUnitModel } from '../../models/document-units/document-unit.model'; 

@Injectable()
export class DocumentUnitMapper implements BaseMapper {

    mapFromJson(json: any): DocumentUnitModel{

        if (!json) {
            return null;
        }

        let model: DocumentUnitModel = new DocumentUnitModel();
        model.id = json.Id;
        model.name = json.Name;
        model.title = json.Title;
        model.subject = json.Subject;
        model.environment = json.Environment;
        model.documentUnitName = json.DocumentUnitName;
        model.year = json.Year;
        model.number = json.Number;

        return model;
    }

}